src/Umbraco.Core/Migrations/Upgrade/V_8_7_0/ConvertToElements.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core.Persistence;
+using Umbraco.Core.Persistence.Dtos;
+
+namespace Umbraco.Core.Migrations.Upgrade.V_8_7_0
+{
+    public class ConvertToElements : MigrationBase
+    {
+        public ConvertToElements(IMigrationContext context) : base(context)
+        {
+        }
+
+        public override void Migrate()
+        {
+            // Get all document type IDs by alias
+            var docTypes = Database.Fetch<ContentTypeDto>();
+            var docTypeMap = new Dictionary<string, int>(docTypes.Count);
+            docTypes.ForEach(d => docTypeMap[d.Alias] = d.NodeId);
+
+            // Find all Nested Content or Block List data types
+            var dataTypes = GetDataTypes(Constants.PropertyEditors.Aliases.BlockList);
+
+            // Find all document types listed in each
+            var elementTypeIds = dataTypes.SelectMany(d => GetDocTypeIds(d.Configuration, docTypeMap)).ToList();
+
+            // Find all compositions those document types use
+            var parentElementTypeIds = Database.Fetch<ContentType2ContentTypeDto>(Sql()
+                .Select<ContentType2ContentTypeDto>()
+                .From<ContentType2ContentTypeDto>()
+                .WhereIn<ContentType2ContentTypeDto>(c => c.ChildId, elementTypeIds)
+                ).Select(c => c.ParentId);
+
+            elementTypeIds = elementTypeIds.Union(parentElementTypeIds).ToList();
+
+            // Convert all those document types to element type
```

Comment:
Is there any chance this can break people's sites? Like if potentially they were using a doc type as both an element type and a doc type?

Reply:
There is, and I thought about that further when I was writing the migrations that I did for #7939 where I decided that if there were any element types already (i.e. this wasn't a v7 to v8 migration), then I wouldn't mess with retyping any document types.  I'll port that change back to here, to minimize the effect on anyone upgrading from a previous v8 version.  We are dealing just with those upgrading Stacked Content, so likely they are still coming straight from v7 anyway, but it would be a good quick check.

Reply:
Cool so since this is already merged i supposed we will just wait until you have a little bit of time to push a new PR, just let us know :) I'll add some notes to the code.

Reply:
Just as a follow-up to this, I ended up pulling the ConvertToElement migration in PR #8336 since the NestedContent portion is PR'd in #7957 and without the StackedContent to BlockEditorList migration, there is nothing else to convert.